pkg/reconcilers/shared/postcreate_hook.go

```diff
@@ -106,9 +109,7 @@ func (r *BaseReconciler) ReconcileUpdatePostCreateHook(ctx context.Context, hcp
 		// Get hook definition
 		pch := &v1alpha1.PostCreateHook{}
 		if err := r.Client.Get(ctx, client.ObjectKey{Name: hookName}, pch); err != nil {
-			errs = append(errs, fmt.Errorf("failed to get PostCreateHook %s: %w", hookName, err))
-			allHooksApplied = false
-			continue
+			return fmt.Errorf("%w: %v", ErrPostCreateHookNotFound, err)
```

Comment:
FYI, https://pkg.go.dev/fmt@go1.24.5#Errorf says that you can use multiple `%w`.